#! /bin/sh
patch -p1 -f $* < $0
exit $?

2008-04-25  Marcus Brinkmann  <marcus@g10code.de>

	* Fix a couple of bugs in W32 GNUTLS support.

diff -rup claws-mail-3.4.0-orig/src/addrindex.h claws-mail-3.4.0/src/addrindex.h
--- claws-mail-3.4.0-orig/src/addrindex.h	2008-04-18 12:47:07.000000000 +0200
+++ claws-mail-3.4.0/src/addrindex.h	2008-04-25 21:43:36.000000000 +0200
@@ -91,6 +91,11 @@ struct _AddressInterface {
 	void (*stopSearch)( void * );
 };
 
+#ifdef G_OS_WIN32
+/* W32 headers define INTERFACE to "struct".  */
+#undef interface
+#endif
+
 typedef struct _AddressDataSource AddressDataSource;
 struct _AddressDataSource {
 	AddrItemObject obj;
diff -rup claws-mail-3.4.0-orig/src/common/ssl_certificate.c claws-mail-3.4.0/src/common/ssl_certificate.c
--- claws-mail-3.4.0-orig/src/common/ssl_certificate.c	2008-04-18 12:47:08.000000000 +0200
+++ claws-mail-3.4.0/src/common/ssl_certificate.c	2008-04-25 21:34:25.000000000 +0200
@@ -33,16 +33,16 @@
 #include <string.h>
 #endif
 #include <sys/types.h>
-#include <sys/socket.h>
-#include <netinet/in.h>
 #include <stdio.h>
-#include <netdb.h>
 #include <glib.h>
 #include <glib/gi18n.h>
-
 #ifdef G_OS_WIN32
-#include "winsock2.h"
-#endif
+#  include <winsock2.h>
+#else
+#  include <sys/socket.h>
+#  include <netinet/in.h>
+#  include <netdb.h>
+#endif /* G_OS_WIN32 */
 #include "ssl_certificate.h"
 #include "utils.h"
 #include "log.h"
